Assets/HoloToolkit/Utilities/Scripts/BoundsExtensions.cs

```diff
@@ -0,0 +1,456 @@
+
+            camera.WorldToScreenPoint(new Vector3(aabbCenter.x - aabbExtents.x, aabbCenter.y - aabbExtents.y, aabbCenter.z + aabbExtents.z)),
+            camera.WorldToScreenPoint(new Vector3(aabbCenter.x + aabbExtents.x, aabbCenter.y - aabbExtents.y, aabbCenter.z + aabbExtents.z)),
+            camera.WorldToScreenPoint(new Vector3(aabbCenter.x + aabbExtents.x, aabbCenter.y + aabbExtents.y, aabbCenter.z + aabbExtents.z)),
+            camera.WorldToScreenPoint(new Vector3(aabbCenter.x - aabbExtents.x, aabbCenter.y + aabbExtents.y, aabbCenter.z + aabbExtents.z))
+        };
+    }
+
+    /// <summary>
+    /// Returns the rectangle which encloses the specifies 'Bounds' instance in screen space.
+    /// </summary>
+    public static Rect GetScreenRectangle(this Bounds bounds, Camera camera)
+    {
+        // Retrieve the bounds' corner points in screen space
+        Vector2[] screenSpaceCornerPoints = bounds.GetScreenSpaceCornerPoints(camera);
+
+        // Identify the minimum and maximum points in the array
+        Vector3 minScreenPoint = screenSpaceCornerPoints[0], maxScreenPoint = screenSpaceCornerPoints[0];
+        for (int screenPointIndex = 1; screenPointIndex < screenSpaceCornerPoints.Length; ++screenPointIndex)
+        {
+            minScreenPoint = Vector3.Min(minScreenPoint, screenSpaceCornerPoints[screenPointIndex]);
+            maxScreenPoint = Vector3.Max(maxScreenPoint, screenSpaceCornerPoints[screenPointIndex]);
+        }
+
+        // Return the screen space rectangle
+        return new Rect(minScreenPoint.x, minScreenPoint.y, maxScreenPoint.x - minScreenPoint.x, maxScreenPoint.y - minScreenPoint.y);
+    }
+
+    /// <summary>
+    /// Returns the volume of the bounds.
+    /// </summary>
+    /// <param name="bounds"></param>
+    /// <returns></returns>
+    public static float Volume(this Bounds bounds)
+    {
+        return bounds.size.x * bounds.size.y * bounds.size.z;
+    }
+
+    /// <summary>
+    /// Returns bounds that contain both this bounds and the bounds passed in.
+    /// </summary>
+    /// <param name="originalBounds"></param>
+    /// <param name="otherBounds"></param>
+    /// <returns></returns>
+    public static Bounds ExpandToContian(this Bounds originalBounds, Bounds otherBounds)
+    {
+        Bounds tmpBounds = originalBounds;
+
+        tmpBounds.Encapsulate(otherBounds);
+
+        return tmpBounds;
+    }
+
+    /// <summary>
+    /// Checks to see if bounds contains the other bounds completely.
+    /// </summary>
+    /// <param name="bounds"></param>
+    /// <param name="otherBounds"></param>
+    /// <returns></returns>
+    public static bool ContainsBounds(this Bounds bounds, Bounds otherBounds)
+    {
+        return bounds.Contains(otherBounds.min) && bounds.Contains(otherBounds.max);
+    }
+
+    public static bool CloserToPoint(this Bounds bounds, Vector3 point, Bounds otherBounds)
```

Comment:
Please add a summary here! The method name wasn't immediately obvious to me until I read the code itself.